### PR TITLE
feat: faster chart endpoint

### PR DESCRIPTION
**Pull request by owlas**
Contributes to: #5074 

Closes: #5295 

The `useSpaces` hook in the frontend is a major contributor to performance issues. I'm going through and removing them all

Showing the updated modals and `/charts` network requests: 

[image](https://www.loom.com/share/fd3aa4077c5a4b43acbe909e08a5644f)

### CODE REVIEW

**Review by rephus — APPROVED** [review]
It looks good to me, nice query building. 

We should have enough e2e coverage for this

File: packages/backend/src/models/SpaceModel.ts [hunk 1]
```diff
@@ -55,6 +57,60 @@ export class SpaceModel {
         this.database = dependencies.database;
     }
 
+    async find(filters: { projectUuid?: string }): Promise<SpaceSummary[]> {
+        const transaction = Sentry.getCurrentHub()
```
[the remark this answers is not in the document]
Comment (owlas) [reply packages/backend/src/models/SpaceModel.ts:61]: I didn't know about that!

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 2]
```diff
@@ -1637,4 +1638,32 @@ export class ProjectService {
...
+                subject('Project', { organizationUuid, projectUuid }),
+            )
+        ) {
+            throw new ForbiddenError();
+        }
+
+        const spaces = await this.spaceModel.find({ projectUuid });
+        const allowedSpaces = spaces.filter(
+            (space) =>
+                space.projectUuid === projectUuid &&
```
[the remark this answers is not in the document]
Comment (owlas) [reply packages/backend/src/services/ProjectService/ProjectService.ts:1659]: True! This is some standard permission checking I want to move to the casl abilities, so I made it verbose.

File: packages/frontend/src/hooks/useChartSummaries.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { ApiError, ChartSummary } from '@lightdash/common';
+import { useQuery } from 'react-query';
+import { lightdashApi } from '../api';
+
+const getChartSummariesInProject = async (projectUuid: string) => {
```
[the remark this answers is not in the document]
Comment (owlas) [reply packages/frontend/src/hooks/useChartSummaries.ts:5]: This is to distinguish it from the `useSavedCharts` hook. I'm doing the same consistently with `useSpaces` vs. `useSpaceSummaries` on another PR. 

Do you think this is confusing? It just means I didn't need to refactor code

**Pull request merged**